tracer/src/Datadog.Trace/Ci/Agent/CIWriterHttpSender.cs

```diff
@@ -122,13 +125,15 @@ private async Task<int> SendPayloadAsync<T>(EventPlatformPayload payload, Func<I
                 var isFinalTry = retryCount >= retryLimit;
                 Exception exception = null;
 
+                sw.Restart();
                 try
                 {
                     statusCode = await SendPayloadAsync(senderFunc, request, payload, state, isFinalTry).ConfigureAwait(false);
                 }
                 catch (MultipartApiRequestNotSupported mReqEx)
                 {
                     Log.Error(mReqEx, "Error trying to send a multipart request to: {Url}", url.ToString());
+                    TelemetryFactory.Metrics.RecordCountCIVisibilityEndpointPayloadDropped(payload.TelemetryEndpoint);
```

Comment:
This isn't right is it? You haven't dropped the payload here - it could have been successful? 🤔

Reply:
Well I manually throw this error when the underlying transport doesn't support MultipartForm requests. And in that case I drop the payload.
https://github.com/DataDog/dd-trace-dotnet/blob/5b9a0cf58ba7f5a5f9d59f542d983ac878042be7/tracer/src/Datadog.Trace/Ci/Agent/CIWriterHttpSender.cs#L256-L262

This is the type of error that it should never occur...

Reply:
_sigh_ this is GitHub formatting messing with me 🤦‍♂️ It looked like it was _outside_ the `catch` 😅  Never mind!